### PR TITLE
Added a multiclass classification example on CIFAR-10 dataset

### PR DESCRIPTION
**Pull request by y27choi**
### Linked issue(s):
Fixes KOL-2331

### What change does this PR introduce and why?
* Added a multiclass classification example on CIFAR-10 dataset

[CIFAR-10 test on Kolena (default configuration)](https://trunk.kolena.io/kolena/results?modelIds=1566&testSuiteId=3198&workflow=Image+Classification&evaluators=N4IgzAjAnAHCBcoIFYBsqFhmA7AXzyA)
[CIFAR-10 test on Kolena (configuration = threshold @ 0.5)](https://trunk.kolena.io/kolena/results?modelIds=1566&testSuiteId=3198&workflow=Image+Classification&evaluators=N4IgzAjAnAHCBcoIFYBsqFhgFmwXzyA)

### Please check if the PR fulfills these requirements

- [ ] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [ ] Relevant docs have been added / updated

### CODE REVIEW

File: examples/classification/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+First, ensure that the `KOLENA_TOKEN` environment variable is populated in your environment. See our
+[initialization documentation](https://docs.kolena.io/installing-kolena/#initialization) for details.
+
+### Multiclass Classification on CIFAR-10
+
+This project defines two scripts that perform the following operations:
+
+1. [`seed_test_suite.py`](scripts/multiclass/seed_test_suite.py) creates the following test suite:
+
+    - `complete cifar10/test [classification]`: complete CIFAR-10
```
Comment (y27choi) [inline examples/classification/README.md]: Will add a stratification test case in a follow up PR.

File: examples/classification/pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+[tool.poetry]
+name = "classification"
+version = "0.1.0"
+description = "Example Kolena integration for multiclass/binary classification"
+authors = ["Kolena Engineering <eng@kolena.io>"]
+license = "Apache-2.0"
+
+[tool.poetry.dependencies]
+python = ">=3.8,<3.12"
+kolena = { path = "../../../kolena/", develop = true }
```
Comment (y27choi) [inline examples/classification/pyproject.toml]: Requires the latest util changes. TODO: update the version.

File: examples/classification/README.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+The data for this example integration lives in the publicly accessible S3 bucket `s3://kolena-public-datasets`.
+
+First, ensure that the `KOLENA_TOKEN` environment variable is populated in your environment. See our
+[initialization documentation](https://docs.kolena.io/installing-kolena/#initialization) for details.
+
+### Multiclass Classification on CIFAR-10
+
+This project defines two scripts that perform the following operations:
+
+1. [`seed_test_suite.py`](scripts/multiclass/seed_test_suite.py) creates the following test suite:
```
Comment (munkyshi) [inline examples/classification/README.md:26]: small nit for this (and other examples): I'd love to see a shell command that I can just copy to know that there aren't any other flags to run with, even if it's as simple as:
```
poetry run python3 scripts/multiclass/seed_test_suite.py
```

File: examples/classification/classification/workflow.py [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+@dataclass(frozen=True)
+class GroundTruth(BaseGroundTruth):
+    """Ground truth type for Classification workflow."""
+
+    classification: Optional[ClassificationLabel]
+    """The classfication label associated with an image. For binary classification, the negative class is `None`."""
+
+
+@dataclass(frozen=True)
+class Inference(BaseInference):
```
Comment (munkyshi) [inline examples/classification/classification/workflow.py:50]: Do we want an `ignore` field, similar to what got added for 2D OD?

File: examples/classification/classification/evaluator_multiclass.py [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+        n_tn = len([True for gt, inf in classification_pairs if gt != label and inf != label])
+
+        class_level_metrics.append(
+            ClassMetricsPerTestCase(
+                label=label,
+                TP=n_tp,
+                FP=n_fp,
+                FN=n_fn,
+                TN=n_tn,
+                Accuracy=compute_accuracy(n_tp, n_fp, n_fn, n_tn),
```
Comment (munkyshi) [inline examples/classification/classification/evaluator_multiclass.py:97]: Accuracy might be a misleading metric -- if I have 10 classes and I have a wrong inference, that still means that it's an FP for one label, a FN for another, and for the other 8, it counts as a `TN`. That means terrible models would still have an accuracy of > 0.8. Is this standard in the approach?

File: examples/classification/classification/evaluator_multiclass.py [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+                Precision=compute_precision(n_tp, n_fp),
+                Recall=compute_recall(n_tp, n_fn),
+                F1=compute_f1_score(n_tp, n_fp, n_fn),
+                FPR=n_fp / (n_fp + n_tn) if n_fp + n_tn > 0 else 0,
```
Comment (munkyshi) [inline examples/classification/classification/evaluator_multiclass.py:101]: Same comment for FPR as for accuracy -- `n_tn` seems like it can be pretty bloated as a metric

**Comment by gordonhart** [comment]
Are we planning to remove `kolena.classification.multiclass` in favor of this example?

File: examples/classification/README.md
Comment (gordonhart) [inline examples/classification/README.md:1]: Also add bullet to `examples/README.md`

File: examples/classification/README.md (hunk 1 above)
Comment (gordonhart) [reply examples/classification/README.md]: Is the `[classification]` suffix necessary? Would prefer to omit for cleanliness

File: examples/classification/classification/workflow.py [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+    metadata: Metadata = dataclasses.field(default_factory=dict)
+    """The metadata associated with the test sample."""
+
+
+@dataclass(frozen=True)
+class GroundTruth(BaseGroundTruth):
+    """Ground truth type for Classification workflow."""
+
+    classification: Optional[ClassificationLabel]
+    """The classfication label associated with an image. For binary classification, the negative class is `None`."""
```
Comment (gordonhart) [inline examples/classification/classification/workflow.py:46]: Did you end up going with `"label" | None` for binary?

**Review by gordonhart — APPROVED** [review]

**Comment by gordonhart** [comment]
Code is clean and the end result on the platform is super nice!

**Comment by y27choi** [comment]
> Are we planning to remove `kolena.classification.multiclass` in favor of this example?

I think we should. We want to encourage customers to build their own workflow or use a pre-built. Any customers that are using `kolena.classification.multiclass`?

File: examples/classification/classification/workflow.py (hunk 7 above)
Comment (y27choi) [reply examples/classification/classification/workflow.py:46]: No, I dropped `Optional` from `GroundTruth` and you will see that change in a subsequent PR.

File: examples/classification/README.md
Comment (y27choi) [reply examples/classification/README.md:1]: Yup, I will follow up on this in another PR.

File: examples/classification/classification/evaluator_multiclass.py (hunk 5 above)
Comment (y27choi) [reply examples/classification/classification/evaluator_multiclass.py:97]: Correct. It can be a misleading metric in some context but it's hard to drop it as it is still one of the standard metrics in Classification workflow.  For example, sklearn has [`classification_report` tool](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.classification_report.html) that summarizes the classification metrics and it includes `accuracy` as well.

File: examples/classification/classification/workflow.py (hunk 4 above)
Comment (y27choi) [reply examples/classification/classification/workflow.py:50]: Ah I did add it on the latest branch.  Will be added in a subsequent PR!

File: examples/classification/README.md (hunk 3 above)
Comment (y27choi) [reply examples/classification/README.md:26]: Gotcha. Will add this in the next PR!

File: examples/classification/README.md
Comment (y27choi) [reply examples/classification/README.md:1]: Done

File: examples/classification/README.md (hunk 3 above)
Comment (y27choi) [reply examples/classification/README.md:26]: Done!

File: examples/classification/README.md (hunk 1 above)
Comment (y27choi) [reply examples/classification/README.md]: Omitted

**Comment by munkyshi** [comment]
> > Are we planning to remove `kolena.classification.multiclass` in favor of this example?
> 
> I think we should. We want to encourage customers to build their own workflow or use a pre-built. Any customers that are using `kolena.classification.multiclass`?

We shared the code with one customer, but it wasn't ever actively used. The only things ever pushed up were examples that we provided.
